qiita_db/portal.py

```diff
@@ -5,13 +5,66 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+import warnings
 
 from .sql_connection import SQLConnectionHandler
 from .util import convert_to_id
+from .exceptions import QiitaDBError, QiitaDBDuplicateError
 from qiita_core.qiita_settings import qiita_config
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
 
 # Only make functions available for main portal
 if qiita_config.portal == "QIITA":
+    def create_portal(portal, desc):
```

Comment:
@wasade I don't understand the fact that the other objects have notions of common state... I don't see anything in the objects that will point out that notion.

Re alternative implementations. I also disagree on re-visiting this, but we the changes introduced by the transaction object this "alternative" implementation will play a key role, as a lot of the logics that we are doing in qiita_pet is going to move to qiita_db in order to make sure that everything is executed in a single transaction.